RecoTracker/TkSeedGenerator/plugins/JetCoreMCtruthSeedGenerator.cc

```diff
@@ -0,0 +1,580 @@
+        GlobalVector bigClustDir = splitClustDirSet.at(cc);
+
+        const auto& simtracksVector = simtracks.product();
+        const auto& simvertexVector = simvertex.product();
+
+        LocalPoint jetInter(0, 0, 0);
+
+        jet_eta = jet.eta();
+        jet_pt = jet.pt();
+
+        const auto& jetVert = jetVertex;  //trackInfo filling
+
+        std::vector<PSimHit> goodSimHit;
+
+        const GeomDet* globDet =
+            DetectorSelector(2, jet, bigClustDir, jetVertex, tTopo);  //select detector mostly hitten by the jet
+
+        if (globDet == nullptr)
+          continue;
+
+        std::pair<std::vector<SimTrack>, std::vector<SimVertex>> goodSimTkVx;
+
+        if (inclusiveConeSeed) {
+          goodSimTkVx = JetCoreMCtruthSeedGenerator::coreTracksFillingDeltaR(
+              simtracksVector, simvertexVector, globDet, jet, jetVert);
+        } else {
+          std::vector<PSimHit> goodSimHit =
+              JetCoreMCtruthSeedGenerator::coreHitsFilling(simhits, globDet, bigClustDir, jetVertex);
+          goodSimTkVx = JetCoreMCtruthSeedGenerator::coreTracksFilling(goodSimHit, simtracksVector, simvertexVector);
+        }
+        seed_number = goodSimTkVx.first.size();
+        std::cout << "seed number in deltaR cone =" << seed_number << std::endl;
+
+        std::vector<std::array<double, 5>> seedVector =
+            JetCoreMCtruthSeedGenerator::seedParFilling(goodSimTkVx, globDet, jet);
+        std::cout << "seedVector.size()=" << seedVector.size() << std::endl;
+
+        for (uint tk = 0; tk < seedVector.size(); tk++) {
+          for (int pp = 0; pp < 5; pp++) {
+            std::cout << "seed " << tk << ", int par " << pp << "=" << seedVector.at(tk).at(pp) << std::endl;
+          }
+          LocalPoint localSeedPoint = LocalPoint(seedVector.at(tk).at(0), seedVector.at(tk).at(1), 0);
+          double track_theta = 2 * std::atan(std::exp(-seedVector.at(tk).at(2)));
+          double track_phi = seedVector.at(tk).at(3);
+          double pt = 1. / seedVector.at(tk).at(4);
+
+          double normdirR = pt / sin(track_theta);
+          const GlobalVector globSeedDir(
+              GlobalVector::Polar(Geom::Theta<double>(track_theta), Geom::Phi<double>(track_phi), normdirR));
+          LocalVector localSeedDir = globDet->surface().toLocal(globSeedDir);
+
+          int64_t seedid = (int64_t(localSeedPoint.x() * 200.) << 0) + (int64_t(localSeedPoint.y() * 200.) << 16) +
+                           (int64_t(seedVector.at(tk).at(2) * 400.) << 32) + (int64_t(track_phi * 400.) << 48);
+          if (ids.count(seedid) != 0) {
+            std::cout << "seed not removed with DeepCore cleaner" << std::endl;
+          }
+          ids.insert(seedid);
+
+          //seed creation
+          float em[15] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+          em[0] = 0.15 * 0.15;
+          em[2] = 0.5e-5;
+          em[5] = 0.5e-5;
+          em[9] = 2e-5;
+          em[14] = 2e-5;
```

Comment:
actually not mandatory. Only if you want to compare le performance with DeepCore, otherwise different numbers should be set. I write a comment about that